app/models/scenario.rb

```diff
@@ -39,4 +39,8 @@ def unique_agent_ids
   def agents_are_owned
     errors.add(:agents, "must be owned by you") unless agents.all? {|s| s.user == user }
   end
+
+  def self.icons
+    Rails.cache.fetch('icons') {YAML.load_file(File.expand_path('config/icons.yml'))}
```

Comment:
Sadly we can not use `Rails.cache` (which defaults to the FileStore) because the file system on Heroku is read-only.
```
@icons ||= begin
  YAML.load_file(File.expand_path('config/icons.yml'))
end
```
will cache the data in memory.